pkg/controllers/workapplier/availability_tracker.go

```diff
@@ -270,11 +270,12 @@ func trackServiceExportAvailability(curObj *unstructured.Unstructured) (Manifest
 		klog.V(2).InfoS("Still need to wait for ServiceExport to be valid", "serviceExport", svcExportObj, "validCondition", validCond)
 		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
 	}
-
+	// Validate annotation weight. Updating the annotation won't change the object generation,
+	// so the current status is not reliable and need to validate the annotation again here
 	weight, err := objectmeta.ExtractWeightFromServiceExport(&svcExport)
 	if err != nil {
-		klog.V(2).InfoS("ServiceExport has invalid weight", "serviceExport", svcExportObj, "error", err)
-		return ManifestProcessingAvailabilityResultTypeFailed, err
+		klog.Errorf(err.Error(), "ServiceExport has invalid weight", "serviceExport", svcExportObj)
+		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, err
```

Comment:
we should not return the error here. no need to retry.
Sorry, it was missed before :(